main.py

```diff
@@ -102,7 +115,42 @@ def parse_config(config_file):
 def ip_to_int(network):
     """ Takes an IP and returns the unsigned integer encoding of the address """
     return struct.unpack('=L', socket.inet_aton(network))[0]
-    
+
+
+def enrich_event(event):
+    """ Takes the raw event data and enriches by adding process tree metadata """
+    proctree_enriched = []
+    error = ""
+    try:
+        proc = psutil.Process(event.pid)
+        proctree = utils.crawl_process_tree(proc)
+        proctree_enriched = list({"pid": p.pid, "cmdline": " ".join(p.cmdline()), "username":  p.username()} for p in proctree)
+    except Exception as e:
+        error=str(e)
```

Comment:
unrelated nit: `traceback.format_exc` will give a lot more context of the exception that occured.

Reply:
Ack, will update

Reply:
Resolved in c9a17c081f0aa5c3c6d78e63cf1e9c42a7110e11 but hilariously I think the ring buffer makes this more performant as I couldn't actually get it to error with a simple `make docker-run` with the `127/8` filter removed (which used to pretty consistently error due to race condition).

Will knock together a simple unit test for this tomorrow, just going to merge as-is for now.